test/cases/testdata/regexreplace/test-regexreplace-0001.yaml

```diff
@@ -0,0 +1,14 @@
+cases:
```

Comment:
I don’t think you’ll need to be that comprehensive here, but a single test for an entire built-in function seems a bit sparse. How does this work with eg capturing group substitutions? How are errors dealt with?

Reply:
good idea! I've covered applying the pattern to subgroups and the error case. Currently, it returns the err as similar to what happens in other regex functions. please let me know if those are enough or if we need more cases to be written